packages/webpack-cli/lib/webpack-cli.js

```diff
@@ -2326,14 +2326,52 @@ class WebpackCLI {
       return;
     }
 
+    const isCacheEnabled = compiler.compilers
+      ? compiler.compilers.some((compiler) => compiler.options.cache)
+      : compiler.options.cache;
+
+    let needForceShutdown = false;
+
+    if (isCacheEnabled) {
+      const signals = ["SIGINT", "SIGTERM"];
+
+      signals.forEach((signal) => {
+        const listener = () => {
+          if (needForceShutdown || typeof compiler.close !== "function") {
+            process.exit();
+          }
+
+          this.logger.info(
+            "Gracefully shutting down. To force exit, press ^C again. Please wait...",
+          );
+
+          needForceShutdown = true;
+
+          compiler.close(() => {
```

Comment:
Let's move check here, otherwise we will not print log for webpack v4

Reply:
Is there a way to gracefully exit in webpack v4?

Reply:
Just output logger and then run `process.exit()`, `if (typeof compiler.close === "function") { compiler.close(() => { process.exit(); }) } else { process.exit(); }`

Reply:
Okay, will do.

Reply:
Done